### PR TITLE
SNAP bik flag

### PR DESCRIPTION
**Pull request by charlessievers**
**Summary**

Added in flag for snap compute to provide per atom bispectrum descriptors instead of summed bispectrum descriptors.

**Author(s)**

Charlie Sievers - Sandia National Labs - charliesievers@cox.net

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Implementation Notes**

Tested file examples/snap/in.snap.compute and produced the same output.

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [ ] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

### CODE REVIEW

**Review by athomps — CHANGES_REQUESTED** [review]
This looks nice and clean.   The only problem is that the treatment of the last column (reference potential) is not clearly defined in the code or in the documentation.  I suggest making a clear choice e.g. 1) `snap all[irow][lastcol] = 0;` for all b_ik rows 2) same, but with reference energy in first b_ik row 3) Put per-atom reference energy in every b_ik row.  Whatever choice is made should be explicitly documented.

File: doc/src/compute_sna_atom.rst [hunk 1]
```diff
@@ -296,6 +299,14 @@ This option is typically used in conjunction with the *chem* keyword,
 and LAMMPS will generate a warning if both *chem* and *bnormflag*
 are not both set or not both unset.
 
+The keyword *bikflag* determines whether or not to expand the bispectrum
+rows of the global array returned by compute snap. If *bikflag* is set
+to *1* then the bispectrum row, which is typically the per-atom bispectrum
+descriptors :math:`B_{i,k}` summed over all atoms *i* to produce
+:math:`B_k`, becomes bispectrum rows equal to the number of atoms. Thus,
+the resulting bispectrum rows are :math:`B_{i,k}` instead of just
+:math:`B_k`.
```
Comment (athomps) [inline doc/src/compute_sna_atom.rst:308]: Add "In this case, the entries in the final column for these rows are set to zero."

**Review by athomps — APPROVED** [review]
Still need to describe final column in documentation.

File: doc/src/compute_sna_atom.rst (hunk 1 above)
Comment (charlessievers) [reply doc/src/compute_sna_atom.rst]: Done

Comment (akohlmey) [reply doc/src/compute_sna_atom.rst]: @athomps @charlessievers as a reminder. If you both consider this to be reading for merging, you need to assign the PR to me.

**Pull request merged**